zproject/backends.py

```diff
@@ -2091,6 +2091,15 @@ def user_data(self, access_token: str, *args: Any, **kwargs: Any) -> Dict[str, s
 
 @external_auth_method
 class AzureADAuthBackend(SocialAuthMixin, AzureADOAuth2):
+    # The upstream implementation uses the outdated /oauth2/authorize
+    # API (instead of the v2.0 API), which doesn't allow us to authenticate
+    # users with just a personal Microsoft account. v2.0 API is required.
+    # This requires us to override the default URLs to use it as well
+    # as adjust the requested scopes, to match this new API.
+    AUTHORIZATION_URL = "{base_url}/oauth2/v2.0/authorize"
+    ACCESS_TOKEN_URL = "{base_url}/oauth2/v2.0/token"
+    DEFAULT_SCOPE = ["User.Read profile openid email"]
```

Comment:
We may want to contribute this upstream, but I'm  not super sure because the old API does still work (even if in a somewhat janky manner). So I'm not sure if upstream just merging a change like this wouldn't cause some back-compat issues for downstream users of the library by some subtler behavior they may rely on through the change from the `/oauth2/...` to the `/oauth2/v2.0/...` Microsoft API...

Reply:
Maybe try asking `@omab` in their Slack about this? I feel like it could make sense to do a new backend or just change the URLs depending on their preference.